Controller/CoreController.php

```diff
@@ -158,10 +155,25 @@ protected function getSearchHandler()
      */
     protected function getBaseTemplate()
     {
-        if ($this->getRequest()->isXmlHttpRequest()) {
+        if ($this->getCurrentRequest()->isXmlHttpRequest()) {
             return $this->getAdminPool()->getTemplate('ajax');
         }
 
         return $this->getAdminPool()->getTemplate('layout');
     }
+
+    /**
+     * Get the request object from the container.
+     *
+     * @return Request
+     */
+    private function getCurrentRequest()
+    {
+        // NEXT_MAJOR: simplify this when dropping sf < 2.4
```

Comment:
We are already redefining the getRequest from the Symfony Controller, so we need a new method to retrieve the request without getting a deprecation message.

IMO we could mark it as in SonataMedia: https://github.com/sonata-project/SonataMediaBundle/pull/1189/commits/64426a90eac5aa2b667fc57c912928872a1b3d26. Just to make sure we move to Request injected as a method parameter. What do you think @greg0ire ?